# OncofusionPerturbSeq

This workflow assumes you start with the output of Cellranger-atac count for each sample (i.e. lane of 10X chip) separately (i.e. not aggregated by Cellranger). 

FF_ArchR.R: works with Cellranger's tsv files to create ArchR Arrow Files and an ArchR project. Nuclei are quality filtered and genotyping information is used here to identify perturbations in nuclei, a UMAP is created (Fig 1d, 2a), a csv for making the peak heatmap is created (Fig 3), all pairwise comparisons in pseudobulk are made between variants and empty vector (Fig 2c) and other relevant comparisons can be made the same way as in Fig 5c, and motif discovery is performed (Fig 4c,d). 

Motif.R: takes the data frames produced earlier and creates Fig 4c,d 

Clust_analysis.R: takes the cluster information generated by ArchR/Seurat and creates Fig 2a,b

GGAA_analysis.R: First uses ArchR to create BED files of marker peaks for variants of interest and then counts the GGAA/TTCC content at those sites

Downsampling.R: Down samples the number of fusion and empty vector nuclei and calculates metrics with the original dataset like fraction of peaks recovered, pearson/spearman correlation of log2FC values etc for Fig 6 b,c. 
